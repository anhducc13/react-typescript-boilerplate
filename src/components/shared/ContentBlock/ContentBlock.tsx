import React from 'react';
import { Card } from 'antd';
import { CardProps } from 'antd/lib/card';
import classNames from 'classnames';
import './ContentBlock.scss';

export interface ContentBlockProps extends CardProps {
  /**
   * Has divider between title and body?
   */
  hasDivider?: boolean;
}

const ContentBlock: React.FC<ContentBlockProps> = props => {
  const { className, hasDivider, children, ...rest } = props;

  return (
    <Card
      className={classNames({
        'content-block': true,
        'has-divider': hasDivider,
        ...(className && { [className]: true }),
      })}
      bordered={false}
      {...rest}
    >
      {children}
    </Card>
  );
};

export default ContentBlock;
