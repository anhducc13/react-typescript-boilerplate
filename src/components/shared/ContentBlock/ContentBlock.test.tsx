import React from 'react';
import { render } from '@testing-library/react';
import ContentBlock from './ContentBlock';

it('renders correctly', () => {
  const { container } = render(<ContentBlock />);
  expect(container.firstChild).toMatchSnapshot();
});
