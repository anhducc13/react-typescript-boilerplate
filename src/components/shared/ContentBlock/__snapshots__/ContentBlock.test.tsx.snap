// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="ant-card content-block"
>
  <div
    class="ant-card-body"
  />
</div>
`;
